examples/hosts/app-integration/schema-upgrade/src/migrator/migrator.ts

```diff
@@ -131,39 +139,12 @@ export class Migrator extends TypedEventEmitter<IMigratorEvents> implements IMig
             }
             await migratedModel.importData(transformedData);
 
-            // Before attaching, let's check to make sure no one else has already done the migration
-            // To avoid creating unnecessary extra containers.
-            if (migratable.migrationTool.migrationState === "migrated") {
-                this._migrationP = undefined;
-                migratedModel.close();
-                this.takeAppropriateActionForCurrentMigratable();
-                return;
-            }
-
             // TODO: Support retry
-            // TODO: Use TaskManager here to reduce container noise.  Specifically -- all clients should race up to
-            // this point (so they're all prepared to do the migration) but should wait for the task lock before
-            // attempting the attach() to minimize the chance that multiple containers are created on the service.
             const containerId = await createResponse.attach();
 
-            // Again, it could be the case that someone else finished the migration during our attach.
-            // Casting to MigrationState because TS doesn't understand that the state may have changed during the
-            // above await.
-            if (migratable.migrationTool.migrationState as MigrationState === "migrated") {
-                this._migrationP = undefined;
-                migratedModel.close();
-                this.takeAppropriateActionForCurrentMigratable();
-                return;
-            }
-
             // TODO: Support retry
             await migratable.migrationTool.finalizeMigration(containerId);
```

Comment:
Suppose I successfully attach(), but then lose the task before I can finalizeMigration().  If I get the task again, I'll attach() another new container currently -- it would be better if we can retain that already-attached container and just call finalizeMigration().